frontend/src/components/AddWatchlist.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <Modal
+    title="Add Watchlist"
+  >
+    <template #button>
+      <ButtonDefault
+        label="New Watchlist"
+      >
+        <template #before>
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            viewBox="0 0 20 20"
+            fill="currentColor"
+            class="w-6 h-6"
+          >
+            <path
+              fill-rule="evenodd"
+              d="M10 18a8 8 0 100-16 8 8 0 000 16zm1-11a1 1 0 10-2 0v2H7a1 1 0 100 2h2v2a1 1 0 102 0v-2h2a1 1 0 100-2h-2V7z"
+              clip-rule="evenodd"
+            />
+          </svg>
+        </template>
+      </ButtonDefault>
+    </template>
+    <template #content="{close}">
+      <label>
+        <div class="text-gray-400 ml-2 mb-1">
+          Name
+        </div>
+        <input
+          v-model="newWatchlistName"
+          class="rounded w-full"
+          placeholder="Name"
+          type="text"
+          @keypress.enter="createList() && close()"
+        >
+      </label>
+    </template>
+    <template #actions="{close}">
+      <ButtonDefault
+        label="Save"
```

Comment:
Here you need to disable the button when clicked, otherwise we will end up with duplicate Watchlists!

I think you could add a `@clicked` event in the button and let the button itself handle `@click` 

In Button.vue

```
@click = disable && emit(clicked) && enable 
```

Reply:
actually in this case the modal is closed when the button is pressed `@click="createList() && close()"`. But for other situations this might be needed.

Reply:
ah, ok you dont have an `await` here :) Otherwise `close()` would be executed after the `.save()`